### PR TITLE
Chore/test tag dvo

### PR DESCRIPTION
**Pull request by stnmtz**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ dist
 lib-web
 *.db
 *.log
+*.bat
```
Comment (slavistan) [inline .gitignore:6]: That would ignore all files named *\*.bat* or *\*/.sh*. Was that intentional? You can add stuff to your local ignore list *.git/info/exclude*, which is useful for scratchpad and test scripts, if that's what you wanted to do.

File: src/dataViews/DataViewExpander.ts [hunk 2]
```diff
@@ -1265,12 +1265,29 @@ export class DataViewExpander {
     public async processIQLQuery(query: IQLQueryJSON): Promise<ProcessedIQLQueryDVO> {
         const matchedAttributeDTOResult = await this.consumption.attributes.executeIQLQuery({ query });
         const matchedAttributeDVOs = await this.expandLocalAttributeDTOs(matchedAttributeDTOResult.value);
+        let valueType, renderHints, valueHints;
+        // Check if we've the same valueType throughout all requests, if yes, set valueType and renderHints.
```
Comment (slavistan) [inline src/dataViews/DataViewExpander.ts:1269]: *...throughout all **matched attributes*** or ***responses***, right?

File: .gitignore (hunk 1 above)
Comment (jkoenig134) [reply .gitignore:6]: I dont care about the bat thing, but we shouldn't exclude sh files here. We use them for example in the .ci folder.

File: src/dataViews/DataViewExpander.ts [hunk 3]
```diff
@@ -1265,12 +1265,29 @@ export class DataViewExpander {
...
+        // If not, delete valueType and renderHints
+        for (const item of matchedAttributeDVOs) {
+            if (!valueType) {
+                valueType = item.valueType;
+                renderHints = item.renderHints;
+                valueHints = item.valueHints;
+            }
+            if (valueType === item.valueType) continue;
+            valueType = renderHints = valueHints = undefined;
```
Comment (jkoenig134) [inline src/dataViews/DataViewExpander.ts:1278]: You should use two lines for that.

File: src/dataViews/DataViewExpander.ts [hunk 4]
```diff
@@ -1265,12 +1265,29 @@ export class DataViewExpander {
     public async processIQLQuery(query: IQLQueryJSON): Promise<ProcessedIQLQueryDVO> {
         const matchedAttributeDTOResult = await this.consumption.attributes.executeIQLQuery({ query });
         const matchedAttributeDVOs = await this.expandLocalAttributeDTOs(matchedAttributeDTOResult.value);
+        let valueType, renderHints, valueHints;
+        // Check if we've the same valueType throughout all requests, if yes, set valueType and renderHints.
+        // If not, delete valueType and renderHints
+        for (const item of matchedAttributeDVOs) {
```
Comment (jkoenig134) [inline src/dataViews/DataViewExpander.ts:1271]: This behaviour looks Kind of Random.

I understand it as "use the first encountered valueType", but what happens if there are many value types?

Maybe the dvo could contain multiple valueType renderHint and valueHint combinations.

File: .gitignore (hunk 1 above)
Comment (stnmtz) [reply .gitignore:6]: Hmm, afaik *.bat only ignores files in the root folder, not every folder. ci/ shouldn't be a problem here (and my vscode is working accordingly).

I wanted to make sure, that sh and bat files are not committed from the root folder, as we might use it for dev purposes only. I remove the gitignores and we need to discuss that offline. Currently, every repo has its own set of gitignores.

File: src/dataViews/DataViewExpander.ts (hunk 4 above)
Comment (stnmtz) [reply src/dataViews/DataViewExpander.ts:1271]: They way it is now: If there is only one single valueType (which is I guess the primary case), use this valueType for the whole query. If there are multiple returned valueTypes, do not return them. One needs to look into each of the results to get the valueType.

The UI layer would then be able to quickly identify if there is only one valueType within the resultset (valueType of the query is set) or not (valueType is not set -> UI needs to render each of the items specifically).

File: src/dataViews/DataViewExpander.ts (hunk 3 above)
Comment (stnmtz) [reply src/dataViews/DataViewExpander.ts:1278]: You mean three? ;)

File: src/dataViews/DataViewExpander.ts [hunk 5]
```diff
@@ -1266,11 +1266,33 @@ export class DataViewExpander {
...
+                valueType = item.valueType;
+                renderHints = item.renderHints;
+                valueHints = item.valueHints;
+            }
+            if (valueType === item.valueType) continue;
+            valueType = undefined;
+            renderHints = undefined;
+            valueHints = undefined;
+            break;
+        }
```
Comment (jkoenig134) [inline src/dataViews/DataViewExpander.ts:1286]:
```suggestion
        const { valueType, renderHints, valueHints } = matchedAttributeDVOs.every((dvo) => dvo.valueType === matchedAttributeDVOs[0].valueType)
        ? { valueType: matchedAttributeDVOs[0].valueType, renderHints: matchedAttributeDVOs[0].renderHints, valueHints: matchedAttributeDVOs[0].valueHints }
        : { valueType: undefined, renderHints: undefined, valueHints: undefined };
```
what about this? would make a bit clearer that every valueType is compared.

File: test/dataViews/IQLQueryExpanded.test.ts
Comment (jkoenig134) [inline test/dataViews/IQLQueryExpanded.test.ts:1]: typically you name the test after the DVO. Why is this different here?

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**